src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -1555,6 +1555,11 @@ private static TypeDescriptionNode NodeFor(object instance, bool createDelegator
             {
                 Type type = instance.GetType();
 
+                if (type.IsCOMObject)
```

Comment:
Does this detect strongly typed COM objects as well as the __ComObject case?

Reply:
Oh, I see you're just restoring the logic that was present in .NET Framework, carry on.